src/number_theory/unique_factorization.lean

```diff
@@ -0,0 +1,74 @@
+import data.nat.prime
```

Comment:
Could you add a [copyright header and module docs](https://leanprover-community.github.io/contribute/doc.html)? The documentation doesn't need to be too long, something along these lines suffices:

```lean
/-#

# Unique factorization in `ℕ`

This file defines the `unique_factorization_monoid` structure on the natural numbers.

## Main results

 * `nat.irreducible_iff_prime`: a non-unit natural number is only divisible by `1` iff it is prime
 * `nat.unique_factorization_monoid`: the natural numbers have unique factorization into irreducibles
 * `nat.factors_eq`: the multiset of elements of `nat.factors` is equal to the factors
   given by the `unique_factorization_monoid` instance

## Tags

prime factorization, prime factors, unique factorization, unique factors
-/
```